channels/management/commands/set_channel_allow_top.py

```diff
@@ -0,0 +1,40 @@
+"""Management command to set allow_top on channels"""
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import User
+from channels.api import Api
+from channels.models import Channel
+
+
+class Command(BaseCommand):
+    """Sets allow_top on channels"""
+    help = 'Sets allow_top on channels'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--name', nargs='?', action='append')
+        parser.add_argument('--allow-top', action='store', default=True)
+
+    def handle(self, *args, **options):
+        """Sets allow_top on channels"""
+        channels = Channel.objects.all()
+
+        print(options)
+        allow_top = options['allow_top']
+
+        if options['name']:
+            channels.filter(name=options['name'])
+
+        api_user = User.objects.get(username=settings.INDEXING_API_USERNAME)
+        api = Api(api_user)
+
+        self.stdout.write("Found {} channels to update".format(channels.count()))
+
+        for channel in channels:
+            try:
+                api.update_channel(channel.name, allow_top=allow_top)
+                self.stdout.write("Channel '{}': set allow_top={}".format(
+                    channel.name,
+                    allow_top,
+                ))
+            except Exception as exc:  # pylint: disable=broad-except
+                self.stderr.write(str(exc))
```

Comment:
Should we let the exception raise instead? If one up the updates fails the user should be able to rerun the command to try again

Reply:
So, I ran into 404s on my local setup (not quite sure why), but I think I would prefer to have the command partially succeed, because otherwise the command could abort part way through and the user would be unable to complete the command until the errors were fixed.

I'll update this so it also prints the channel name though, because it's not super helpful as-is.

Reply:
That's fine, can we also make it exit with a non-zero code if there are any errors?